flow/src/main/java/flow/Flow.java

```diff
@@ -305,6 +327,14 @@ void enqueue(PendingTraversal pendingTraversal) {
       }
     }
 
+    void bootstrap(History history) {
+      this.nextHistory = checkNotNull(history, "history");
+      if (dispatcher == null) {
+        throw new AssertionError("Bad doExecute method allowed dispatcher to be cleared");
+      }
+      dispatcher.dispatch(new Traversal(null, nextHistory, Direction.REPLACE), this);
```

Comment:
It's never useful to have the same history handed to you in both fields, it's just a weird edge case you have to know how to interpret.

It is often useful to know when you're handling an initial/bootstrap traversal, so you avoid duplicative/pointless work. 

First dispatch after config change looks exactly like cold start, nothing special there.